### PR TITLE
Add a deepset model

### PR DESCRIPTION
**Pull request by jwnys**
NetKet has a very specific DeepSetRelDists model, but does not yet contain a simple DeepSet architecture. I've added a relatively flexible one. I tried to keep DeepSetRelDists untouched.

### CODE REVIEW

**Comment by gcarleo** [comment]
Nice

**Comment by PhilipVinc** [comment]
What do you think about adding a "deepsets" layer that takes as input some rank 3 tensor (multiple channels for the multiple inputs) so that it can be reused more generally, as well as in this MLP + Deepsets architecture?

**Comment by jwnys** [comment]
> 

Well, this essentially does that, but since it's a composition of layers, I'd call it a model. You can require a multidimensional output and put such layers together... The input that we expect is (batch, N, D), and we sum over N to make it invariant. Is this what you have in mind?

File: netket/models/deepset.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 from netket.utils import deprecate_dtype
 from netket.hilbert import ContinuousHilbert
+import netket as nk
```
Comment (gcarleo) [inline netket/models/deepset.py:15]: maybe best to import only MLP from nk.models ?

Comment (PhilipVinc) [reply netket/models/deepset.py:15]: Nice catch. Yeah tou should not import netket inside netket because it might wreak havok in the future

**Comment by jwnys** [comment]
I think these failing tests have nothing to do with my DeepSet.

**Comment by PhilipVinc** [comment]
Please rebase on master to fix CI woes

**Review by PhilipVinc — COMMENTED** [review]
I had reviewed this a long time ago but never 'published' the comments.
It's a bit messy, but as it is a model we could merge it if you fix the issues here.

Also please add a notice in the CHAGELOG

**Review by PhilipVinc — APPROVED** [review]
nice!

**Comment by PhilipVinc** [comment]
thanks for all the effort! I know it's annoying but I think this got to a point that is fairly easy to use and flexible.

**Pull request merged**